### PR TITLE
Reformat database

### LINKED ISSUE #275: Database fixes
- [x] use foreign key for user_id
- [x] use foreign key for goal_id
- [x] change the primary key of a rating to text.

### PR DESCRIPTION
**Pull request by shiryz**
Fixes #275 .
Adds foreign key constraints and changes date from time stamp to text.

### CODE REVIEW

File: server/database/db_build.sql [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
 CREATE TABLE goals (
-  goal_id TEXT PRIMARY KEY NOT NULL,
-  user_id TEXT NOT NULL,
+  goal_id TEXT NOT NULL PRIMARY KEY,
+  user_id TEXT REFERENCES users(user_id),
   title TEXT NOT NULL,
   icon TEXT,
   deleted BOOLEAN DEFAULT FALSE,
-  date_created TIMESTAMP NOT NULL DEFAULT now()
+  date_created TEXT
```
Comment (emilyb7) [inline server/database/db_build.sql]: Not sure that we should convert to text here...

File: server/database/db_build.sql [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-  rating_id SERIAL PRIMARY KEY NOT NULL,
-  user_id TEXT,
-  goal_id TEXT,
+  rating_id TEXT NOT NULL PRIMARY KEY,
+  user_id TEXT REFERENCES users(user_id),
+  goal_id TEXT REFERENCES goals(goal_id),
   rating INTEGER,
   comment TEXT,
-  date_created TIMESTAMP NOT NULL DEFAULT now()
+  date_created TEXT
```
Comment (emilyb7) [inline server/database/db_build.sql]: ... and here

Comment (shiryz) [inline server/database/db_build.sql]: It depends on what moment.js returns, I thought it was a string, if it's still a date then it can be changed

Comment (emilyb7) [inline server/database/db_build.sql]: it returns a string, but TIMESTAMP is also converted to a string in javascript so it's ok!

Comment (emilyb7) [inline server/database/db_build.sql]: at least i think so... it all seems to be fine at the moment. Whatever I said before might have been wrong...

Comment (shiryz) [inline server/database/db_build.sql]: I've checked moment.js and if we want a timestamp we're going to have to transform what we get to date

**Pull request merged**